src/tests/base_kubeapi_test.py

```diff
@@ -150,12 +151,42 @@ def start_nodes(cls, nodes: Nodes, infra_env: InfraEnv, entity_config: BaseEntit
 
         log.info("waiting for host agent")
         agents = infra_env.wait_for_agents(len(nodes))
+        node_list = nodes.controller.list_nodes()
         for agent in agents:
             agent.approve()
-            cls.set_agent_hostname(nodes[0], agent, entity_config.is_ipv4)  # Currently only supports single node
+            cls._set_host_name_from_node(node_list, agent, entity_config.is_ipv4)
 
         return agents
 
+    """
+    Use the MAC address that is listed in the virt node object to find the interface entry
+    in the host's inventory and take the host name from there
+    The setting of the hostname is required for IPv6 only, because the nodes are booted with
+    hostname equal to localhost which is neither unique not legal name for AI host
+    """
+
+    @classmethod
+    def _set_host_name_from_node(cls, nodes: List[Node], agent: Agent, is_ipv4: bool) -> None:
+        if is_ipv4:
```

Comment:
```suggestion
    @classmethod
    def _set_host_name_from_node(cls, nodes: List[Node], agent: Agent, is_ipv4: bool) -> None:
        """
    Use the MAC address that is listed in the virt node object to find the interface entry
    in the host's inventory and take the host name from there
    The setting of the hostname is required for IPv6 only, because the nodes are booted with
    hostname equal to localhost which is neither unique not legal name for AI host
    """

        if is_ipv4:
```